### PR TITLE
checking for hotkey.txt in miner's HF repo for the contents of their …

### PR DESCRIPTION
**Pull request by xzistance**
…associated hotkey. if hotkey being scored doesn't match, score 0

### CODE REVIEW

File: miner_utils/upload_model.py [hunk 1]
```diff
@@ -152,14 +153,14 @@ async def main(config: bt.config):
     # We can only commit to the chain every n minutes, so run this in a loop, until successful.
     while True:
         try:
+            await model_metadata_store.store_model_metadata(
+                wallet.hotkey.ss58_address, model_id_with_hash
+            )
```
Comment (square-1111) [inline miner_utils/upload_model.py:158]: Why is this part moved here?

**Review by square-1111 — APPROVED** [review]

File: miner_utils/upload_model.py (hunk 1 above)
Comment (xzistance) [reply miner_utils/upload_model.py:158]: I think the idea was to not make the repo public until after posting the repo info to the chain to further assist with trying to evade the copycats

File: neurons/model_scoring.py [hunk 2]
```diff
@@ -107,7 +119,16 @@ def embed_text(imagebind, texts: List[str], device) -> List[torch.FloatTensor]:
 def get_model_score(hf_repo_id, mini_batch, local_dir, hotkey, block, model_tracker):
     cleanup_gpu_memory()
     log_gpu_memory('before model load')
-    inference_recipe, config = load_ckpt_from_hf(hf_repo_id, local_dir)
+    inference_recipe, config, hotkey_file_contents = load_ckpt_from_hf(hf_repo_id, local_dir)
+
+    # Check if the contents of license file are the same as the hotkey if in repo
+    if hotkey_file_contents is not None and hotkey_file_contents != hotkey:
```
Comment (salmanshah1d) [inline neurons/model_scoring.py:125]: shouldn't this be `if hotkey_file_contents is None or hotkey_file_contents != hotkey` so that if it's None we don't just give it a free pass?

**Review by salmanshah1d — APPROVED** [review]
